plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -1559,23 +1559,47 @@ function woocommerce_result_count() {
 
 	/**
 	 * Output the product sorting options.
+	 *
+	 * @param array|null $attributes Block attributes.
 	 */
-	function woocommerce_catalog_ordering() {
+	function woocommerce_catalog_ordering( $attributes = null ) {
 		if ( ! wc_get_loop_prop( 'is_paginated' ) || ! woocommerce_products_will_display() ) {
 			return;
 		}
 		$show_default_orderby    = 'menu_order' === apply_filters( 'woocommerce_default_catalog_orderby', get_option( 'woocommerce_default_catalog_orderby', 'menu_order' ) );
-		$catalog_orderby_options = apply_filters(
-			'woocommerce_catalog_orderby',
-			array(
-				'menu_order' => __( 'Default sorting', 'woocommerce' ),
-				'popularity' => __( 'Sort by popularity', 'woocommerce' ),
-				'rating'     => __( 'Sort by average rating', 'woocommerce' ),
-				'date'       => __( 'Sort by latest', 'woocommerce' ),
-				'price'      => __( 'Sort by price: low to high', 'woocommerce' ),
-				'price-desc' => __( 'Sort by price: high to low', 'woocommerce' ),
-			)
-		);
+
+		if ( isset( $attributes ) && isset( $attributes['useLabel'] ) && $attributes['useLabel'] ) {
+			/**
+			 * Filters the catalog orderby options.
+			 *
+			 * @since 9.7.0
+			 * @param array $catalog_orderby_options Array of catalog orderby options.
+			 */
+			$catalog_orderby_options = apply_filters(
+				'woocommerce_catalog_orderby',
+				array(
+					'menu_order' => __( 'Default', 'woocommerce' ),
+					'popularity' => __( 'Popularity', 'woocommerce' ),
+					'rating'     => __( 'Average rating', 'woocommerce' ),
+					'date'       => __( 'Latest', 'woocommerce' ),
+					'price'      => __( 'Price: low to high', 'woocommerce' ),
+					'price-desc' => __( 'Price: high to low', 'woocommerce' ),
+				)
+			);
+		} else {
+			// phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingHookComment
+			$catalog_orderby_options = apply_filters(
+				'woocommerce_catalog_orderby',
+				array(
+					'menu_order' => __( 'Default sorting', 'woocommerce' ),
+					'popularity' => __( 'Sort by Popularity', 'woocommerce' ),
+					'rating'     => __( 'Sort by Average rating', 'woocommerce' ),
+					'date'       => __( 'Sort by Latest', 'woocommerce' ),
+					'price'      => __( 'Sort by Price: low to high', 'woocommerce' ),
+					'price-desc' => __( 'Sort by Price: high to low', 'woocommerce' ),
+				)
+			);
```

Comment:
Can we please avoid changing first letter case for previous labels? For example, I can see that "popularity" is changed to "Popularity". As there might be existing plugins that may rely on exact string matching. This could break plugin compatibility.

Reply:
Good catch!
Fixed in [Preserve first letter as lowercase](https://github.com/woocommerce/woocommerce/pull/52838/commits/25fe4259d1bb2c2177851051c6fac445ff6a8bed)